### DISCUSSION TITLE
Image assets and playing nicely with Cloudflare

### DISCUSSION THREAD

**Discussion by schwartzmj** [statamic/cms#6487] [Q&A]
I'm trying to get Statamic to play as nice as possible with Cloudflare re: assets.

Static assets work great and get cached easily enough. The issue I'm having is with Glide generated images.

Since there are no file extensions on these image paths, Cloudflare by default will not cache them. While this isn't perfect for this scenario, it's fine. You can just set up a Page Rule in Cloudflare to cache everything in the path of `/img` — except you can't.

Laravel / Statamic send back cookies at these image routes, which Cloudflare will never cache. I tested playing around with Middleware on the Laravel side of things and playing with Cloudflare Workers on that side, but wasn't able to figure it out.

I'm no Laravel genius so I could be missing something here. Any ideas for this?

**Comment by jasonvarga** [comment]
If you use the "cached" setting, there will be extensions.
https://statamic.dev/image-manipulation#custom-path-static

**Reply by schwartzmj** [reply]
Sorry, I should have mentioned this in my original message. 🤦 

My understanding is that method has a downside in that it generates all image sizes on page load, even if they never get used. With that said I may test it again and see if it works for us.

Would https://statamic.dev/image-manipulation#custom-disk-cdn potentially help with this? I'll probably test this out and see. I must have missed it before.

Anyways, is there any other way to get around this without that flag? Either by not sending cookies on those routes or having Glide/Statamic add the file extensions? I realize the latter could potentially be a big undertaking, but I'm just trying to figure out what's possible before diving in too deep. My motivation is part "actually trying to find a clean solution" and part just plain curiosity.

**Reply by jasonvarga** [reply]
Yeah it'll do everything on page load. Was just giving a workaround.

You're saying that if there's a file extension Cloudflare will be okay to cache it?
Or, if there are no cookies?

Either option would potentially be doable. We could let probably let Glide output file extensions on the urls, then we could just ignore them.
Or, we could move the img routes out of the web middleware group. I don't think they actually _need_ to be there. There's no authentication on those routes.

**Reply by schwartzmj** [reply]
To directly answer your question, basically(-ish) both.

There's two things here regarding Cloudflare.

1. By default, they cache based on file extension:
> Cloudflare only caches based on file extension and not by MIME type. The Cloudflare CDN does not cache HTML by default. Additionally, Cloudflare caches a website’s robots.txt.

I can tell them to "Cache Everything" under the `/img/` path, however issue 2 comes up:

2. They do not cache responses containing cookies.

I *think* just moving them out of the web middleware group would allow the `/img` path thing to work. *Ideally* though I'd personally like to **also** have the file extension appended so that it "just works" without that additional `/img` path configuration on the Cloudflare end.

Last night when I was tinkering I was trying to get Laravel/Statamic to remove the web middleware from the `/img` path but couldn't figure it out. Me dumb.

Here's the page I quoted for reference: https://developers.cloudflare.com/cache/about/default-cache-behavior/
